### PR TITLE
[v2.0.0] Add Jest unit test runner and hooks

### PR DESCRIPTION
**Pull request by phamann**
**⚠️ Note this is a PR into the v.2.0.0 branch and not master**

### TL;DR
Adds [`Jest`](https://jestjs.io) testing framework dependencies to run unit tests within the project. We also add `test` and `test:watch` npm scripts which executes the tests with Jest via Babel with the option of running them in watch mode. Tests and linting are also enforced via a pre-push git hook using [`Husky`](https://github.com/typicode/husky).

### Note
- To ensure we have parity with the source files within the project, we are writing are unit tests in TypeScript and transpiling them with babel before passing to Jest using the `babel-jest` preset.
- We set coverage to be reported to stdout and output to a `coverage` directory by default. 
- I have included to example tests in this PR for the woker.ts file

### CODE REVIEW

File: src/worker.spec.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import * as worker from "./worker";
+
+describe("worker", (): void => {
```
Comment (clight-fastly) [inline src/worker.spec.ts]: TS assumes `void` if you don't return anything so you can drop the explicit `void` return type from these funcs if you like

File: package.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
   "main": "dist/lib.js",
   "license": "MIT",
   "scripts": {
-    "test": "",
+    "test": "jest --coverage",
```
Comment (clight-fastly) [inline package.json]: May want to switch this command to `jest` and make `test` do something like `npm jest && npm lint` (since lint runs tsc)

Comment (clight-fastly) [reply package.json]: If you want paralellism for the `test` command a quick tool to use is `concurrently`
https://github.com/kimmobrunfeldt/concurrently

**Comment by oso-mate** [comment]
I didn't know much about Jest, would anyone like to briefly introduce it in the first Frontend Salon?

File: src/worker.spec.ts (hunk 1 above)
Comment (phamann) [reply src/worker.spec.ts]: I dug into this, even though our `tsconfig.json` doesn't set it, I am getting warnings from the ESLint TypeScript plugins for `"explicit-function-return-type": "warn"`. Hence why I've been declaring the return type. https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/recommended.json#L10
